sky/backends/cloud_vm_ray_backend.py

```diff
@@ -1722,16 +1745,25 @@ def provision_with_retries(
                 num_nodes = task.num_nodes
                 cluster_exists = False
 
-            # Set to None so that sky.optimize() will assign a new one
-            # (otherwise will skip re-optimizing this task).
-            # TODO: set all remaining tasks' best_resources to None.
-            task.best_resources = None
-            self._dag = sky.optimize(self._dag,
-                                     minimize=self._optimize_target,
-                                     blocked_resources=self._blocked_resources)
-            to_provision = task.best_resources
-            assert task in self._dag.tasks, 'Internal logic error.'
-            assert to_provision is not None, task
+            while True:
+                # Set to None so that sky.optimize() will assign a new one
+                # (otherwise will skip re-optimizing this task).
+                # TODO: set all remaining tasks' best_resources to None.
+                task.best_resources = None
+                self._dag = sky.optimize(
+                    self._dag,
+                    minimize=self._optimize_target,
+                    blocked_resources=self._blocked_resources)
+                to_provision = task.best_resources
+                assert task in self._dag.tasks, 'Internal logic error.'
+                assert to_provision is not None, task
+
+                # Skip clouds that do not support requested features.
+                if to_provision.cloud.support(self._requested_features):
+                    break
+                self._blocked_resources.add(
+                    resources_lib.Resources(cloud=to_provision.cloud))
+
```

Comment:
We don't need the changes here by adopting the changes in the comment above. : )